src/cluster_copier.cc

```diff
@@ -330,14 +330,21 @@ namespace grpc_labview {
             {
                 auto nestedMetadata = repeatedNested->_value.front()->_metadata;
                 auto clusterSize = nestedMetadata->clusterSize;
+                auto byteSize = repeatedNested->_value.size() * clusterSize;
+                auto alignment = nestedMetadata->alignmentRequirement;
+                auto alignedElementSize = byteSize / alignment;
+                if (byteSize % alignment != 0)
+                {
+                    alignedElementSize++;
+                }
 
-                NumericArrayResize(0x08, 1, start, repeatedNested->_value.size() * clusterSize);
+                NumericArrayResize(GetTypeCodeForSize(alignment), 1, start, alignedElementSize);
                 auto array = *(LV1DArrayHandle*)start;
                 (*array)->cnt = repeatedNested->_value.size();
```

Comment:
Your understanding is more or less correct. I wouldn't consider myself to be that knowledgeable on the nuances between using `NumericArrayResize` vs. `DSSetHandleSize`. I suspect you might need to use [DSNewHandle](https://www.ni.com/docs/en-US/bundle/labview-api-ref/page/properties-and-methods/lv-manager/dssethandlesize.html?srsltid=AfmBOooRebmUqvdTjIppBAtJYKkAYS4w1Gzm_hVDj9I6GNgTaZkop3cf) as I'm not sure if `DSSetHandleSize` can only resize a valid handle vs. also allocating a new handle. It's also unclear how much alignment logic is present in those functions vs. `NumericArrayResize`. The documentation for `NumericArrayResize` states:

> Resizes a data handle that refers to a numeric array. This routine also accounts for alignment issues. It does not set the array dimension field. If *dataHP is NULL, LabVIEW allocates a new array handle in *dataHP.

The documentation for the other memory manager functions does no't explicitly state this so I'm not sure how much the methods will do for you and how much you have to figure out for yourself. #352 included changes which explicitly changed from `DSSetNewHandle` to `NumericArrayResize`, but I'm not entirely sure why. I had enough other things to worry about that I didn't bother poking into this further.

Reply:
> I suspect you might need to use [DSNewHandle](https://www.ni.com/docs/en-US/bundle/labview-api-ref/page/properties-and-methods/lv-manager/dssethandlesize.html?srsltid=AfmBOooRebmUqvdTjIppBAtJYKkAYS4w1Gzm_hVDj9I6GNgTaZkop3cf) as I'm not sure if DSSetHandleSize can only resize a valid handle vs. also allocating a new handle.

That's what I meant by "It takes a handle rather than an hp..." If you want the same behavior as NumericArrayResize where you can start with a null handle, you have to write something like `if (h) DSSetHandleSize(h, ...) else h = DSNewHandle(...)`. 

> It's also unclear how much alignment logic is present in those functions vs. NumericArrayResize.

[DSNewHandle](https://www.ni.com/docs/en-US/bundle/labview-api-ref/page/properties-and-methods/lv-manager/dsnewhandle.html) says

> The routine aligns all handles and pointers in DS to accommodate the largest possible data representations for the platform in use.

I checked the source and this is at least 8.

Anyway, I don't see any problems with this use of NumericArrayResize.